test/MinimalDelegation.execute.t.sol

```diff
@@ -628,4 +628,103 @@ contract MinimalDelegationExecuteTest is TokenHandler, HookHandler, ExecuteFixtu
         signerAccount.execute(signedBatchedCall, signature);
         assertEq(Settings.unwrap(signerAccount.getKeySettings(newKey.toKeyHash())), 0);
     }
+
+    function test_multicall_succeeds() public {
+        // Create two separate batches of calls
+        Call[] memory calls1 = CallUtils.initArray();
+        Call[] memory calls2 = CallUtils.initArray();
+
+        calls1 = calls1.push(buildTransferCall(address(tokenA), address(receiver), 1e18)); // Transfer 1 tokenA
+        calls2 = calls2.push(buildTransferCall(address(tokenB), address(receiver), 1e18)); // Transfer 1 tokenB
+
+        TestKey memory p256Key1 = TestKeyManager.initDefault(KeyType.P256);
+        TestKey memory p256Key2 = TestKeyManager.initDefault(KeyType.P256);
+
+        vm.startPrank(address(signer));
+        signerAccount.register(p256Key1.toKey());
+        signerAccount.register(p256Key2.toKey());
+        vm.stopPrank();
+
+        // Create signed calls for each batch
+        BatchedCall memory batchedCall1 = CallUtils.initBatchedCall().withCalls(calls1).withShouldRevert(true);
+        SignedBatchedCall memory signedBatchedCall1 = CallUtils.initSignedBatchedCall().withBatchedCall(batchedCall1)
+            .withNonce(DEFAULT_NONCE).withKeyHash(p256Key1.toKeyHash());
+        bytes memory signature1 = p256Key1.sign(signerAccount.hashTypedData(signedBatchedCall1.hash()));
+
+        BatchedCall memory batchedCall2 = CallUtils.initBatchedCall().withCalls(calls2).withShouldRevert(true);
+        SignedBatchedCall memory signedBatchedCall2 = CallUtils.initSignedBatchedCall().withBatchedCall(batchedCall2)
+            .withNonce(DEFAULT_NONCE + 1).withKeyHash(p256Key2.toKeyHash());
+        bytes memory signature2 = p256Key2.sign(signerAccount.hashTypedData(signedBatchedCall2.hash()));
+
+        // Create multicall data
+        bytes[] memory multicallData = new bytes[](2);
+        multicallData[0] = abi.encodeWithSelector(
+            bytes4(keccak256("execute((((address,uint256,bytes)[],bool),uint256,bytes32),bytes)")),
```

Comment:
nit you should use a test builder library for this, we should have something in either CallUtils or otherwise